graviti/dataframe/sql/series.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""The implementation of the search related Series."""
+
+from typing import Any, Callable, Dict, Type, TypeVar, Union
+
+import graviti.portex as pt
+from graviti.dataframe.sql.container import SearchContainer, SearchContainerRegister
+
+_S = TypeVar("_S", bound="SeriesBase")
+
+
+class SeriesBase(SearchContainer):
+    """The base class for the sql Series.
+
+    Arguments:
+        expr: The expression of the search.
+        schema: The schema of the series.
+
+    """
+
+    operators: Dict[str, str] = {}
+
+    def __init__(  # pylint: disable=super-init-not-called
+        self,
+        expr: Union[str, Dict[str, Any]],
+        schema: pt.PortexType,
+    ) -> None:
+        self.expr = expr
+        self.schema = schema
+
+    def __eq__(self, right: Any) -> "BooleanSeries":  # type: ignore[override]
+        if isinstance(right, SeriesBase):
+            expr = {"$eq": [self.expr, right.expr]}
+        else:
+            expr = {"$eq": [self.expr, right]}
+
+        return BooleanSeries(expr, pt.boolean())
```

Comment:
the type of BooleanSeries must be pt.boolean(), do we need to set type as a parameter?